pkg/kubelet/kubelet.go

```diff
@@ -1087,6 +1088,7 @@ func (kl *Kubelet) shouldContainerBeRestarted(pod *api.BoundPod, containerName,
 	}
 
 	if len(recentContainers) > 0 {
+		dockerContainerName := dockertools.BuildDockerName(pod.UID, podFullName, container)
 		if pod.Spec.RestartPolicy.Never != nil {
 			glog.Infof("Already ran container with name %s, do nothing",
```

Comment:
I would say to just use the podFullName instead. The Docker container name is not building us anything and confusing the code.